tests/components/vacuum/test_mqtt.py

```diff
@@ -23,33 +19,46 @@ def setUp(self):  # pylint: disable=invalid-name
         self.hass = get_test_home_assistant()
         self.mock_publish = mock_mqtt_component(self.hass)
 
+        self.default_config = {
+            CONF_PLATFORM: 'mqtt',
+            CONF_NAME: 'mqtttest',
+            mqtt.CONF_COMMAND_TOPIC = 'vacuum/command'
+            mqtt.CONF_SEND_COMMAND_TOPIC = 'vacuum/send_command'
+            mqtt.CONF_BATTERY_LEVEL_TOPIC = 'vacuum/state'
+            mqtt.CONF_BATTERY_LEVEL_TEMPLATE = '{{ value_json.battery_level }}'
+            mqtt.CONF_CHARGING_TEMPLATE = '{{ value_json.charging }}'
+            mqtt.CONF_CLEANING_TEMPLATE = '{{ value_json.cleaning }}'
+            mqtt.CONF_DOCKED_TEMPLATE = '{{ value_json.docked }}'
+            mqtt.CONF_STATE_TOPIC = 'vacuum/state'
+            mqtt.CONF_STATE_TEMPLATE = '{{ value_json.state }}'
+            mqtt.CONF_FAN_SPEED_TEMPLATE = '{{ value_json.fan_speed }}'
+            mqtt.CONF_SET_FAN_SPEED_TOPIC = 'vacuum/set_fan_speed'
+        }
+
     def tearDown(self):  # pylint: disable=invalid-name
         """Stop down everything that was started."""
         self.hass.stop()
 
     def test_default_supported_features(self):
         """Test that the correct supported features."""
         self.assertTrue(setup_component(self.hass, vacuum.DOMAIN, {
-            vacuum.DOMAIN: {
-                CONF_PLATFORM: 'mqtt',
-                CONF_NAME: 'mqtttest',
-            }
+            vacuum.DOMAIN: self.default_config,
         }))
         entity = self.hass.states.get('vacuum.mqtttest')
-        entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
-        self.assertListEqual(sorted(services_to_strings(entity_features)),
+        entity_features = \
+            entity.attributes.get(mqtt.CONF_SUPPORTED_FEATURES, 0)
+        self.assertListEqual(sorted(mqtt.services_to_strings(entity_features)),
                              sorted(['turn_on', 'turn_off', 'stop',
                                      'return_home', 'battery', 'status',
                                      'clean_spot']))
 
     def test_all_commands(self):
         """Test simple commands to the vacuum."""
         self.assertTrue(setup_component(self.hass, vacuum.DOMAIN, {
-            vacuum.DOMAIN: {
-                CONF_PLATFORM: 'mqtt',
-                CONF_NAME: 'mqtttest',
-                ATTR_SUPPORTED_FEATURES: services_to_strings(ALL_SERVICES),
-            }
+            vacuum.DOMAIN: self.default_config.update({
```

Comment:
Neat, great to have these shared defaults from `self.default_config`. Thanks for the cleanup